### PR TITLE
[addon] add doxygen documentation for skin controls

### PR DESCRIPTION
**Pull request by AlwinEsch**
Here the next final part of my development kit.

It add the documentation of skin controls taken from http://kodi.wiki/view/Controls
Is up do date with the wiki.

Is visible here: http://alwinesch.github.io

### CODE REVIEW

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by Razzeee** [comment]
@phil65 @ronie @HitcherUK @BigNoid @vonH

**Comment by AlwinEsch** [comment]
Thanks @Razzeee 

have updated and fixed few parts, also added the first description on "Skin Development" start website.

Demo Website with link above is updated and have now the original from this request.

**Comment by ronie** [comment]
i'm not sure i understand the reasoning behind this...
what's the purpose of duplicating the entire skinning manual in the source code?

**Comment by AlwinEsch** [comment]
Main reason about them is to add a reworked version of the python documentation and also add it for binary add-ons step by step.

The parts like on this request are needed for add-ons and as reference on related documents.

With the wiki is OK, but have a bad overview of documents, also is it with this more easy to do document updates direct with code changes (must only be looked about them on requests).

Maybe we can remove the main development part complete from wiki in future and use the doxygen way about them.

File: xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/controls/Fade_Label_Control.txt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+--------------------------------------------------------------------------------
+\section Fade_Label_Control_sect3 See also
+
+#### Development:
+
+- [Add-on development](http://kodi.wiki/view/Add-on_development)
+- [Skinning](http://kodi.wiki/view/Skinning)
+
+*/
```
Comment (AlwinEsch) [inline xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/controls/Fade_Label_Control.txt]: Also is a text like this (for me) more human read- and writable as on wiki http://kodi.wiki/index.php?title=Fade_label_control&action=edit

File: xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/controls/RSS_feed_Control.txt [hunk 2]
```diff
@@ -0,0 +1,106 @@
+/*!
+
+\page RSS_feed_Control RSS ticker
+\brief **Used to display scrolling RSS feeds.**
+
+\tableofcontents
+
+Kodi can display an RSS feed on the home screen of the default skin/interface
+(Confluence), as well as any other skin that supports RSS feeds. By default,
```
Comment (crash123) [inline xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/controls/RSS_feed_Control.txt]: Estuary?

File: xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/skin.dox [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+Kodi, from the images, the sizes and positions of all controls, colours, fonts,
+and text, through to altering navigation and even adding new functionality.
+
+The skin system is quite complex, and this portion of the manual is dedicated to
+providing in depth information on how it all works, along with tips to make the
+experience a little more pleasant.
+
+If you are just getting started with skinning Kodi, then it is suggested that the
+best way to learn is by modifying one of the many existing skins that are
+available. The default skin, Confluence, includes almost all the various tricks
```
Comment (crash123) [inline xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/skin.dox]: Estuary?

Comment (AlwinEsch) [inline xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/skin.dox]: Upps, c&p fault.

What think you about:
"The default skin, Estuary (or the old Confluence), includes almost all the various tricks"
?

Or is the best only the Estuary alone?

Comment (phil65) [inline xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/skin.dox]: just  "The default skin "Estuary" includes almost all the various tricks" is fine.

File: xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/skin.dox [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+\brief \htmlonly
+      <h3><span  style="text-decoration: underline;"><span  style="font-style: italic;"><span
+               style="color: rgb(102, 102, 102);">Skin Add-On Development</span></span></span></h3>
+       \endhtmlonly
+
+Kodi includes a GUI library that allows you to skin/change everything you see in
+Kodi, from the images, the sizes and positions of all controls, colours, fonts,
+and text, through to altering navigation and even adding new functionality.
+
+The skin system is quite complex, and this portion of the manual is dedicated to
```
Comment (crash123) [inline xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/skin.dox]: Just a suggestion to break up your long sentences: The skin system is quite complex. This portion of the manual is dedicated to providing in depth information on how it all works. It contains tips to make the experience a little more pleasant.

File: xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/skin.dox [hunk 5]
```diff
@@ -0,0 +1,27 @@
+/*!
+
+\page skin_parts Skin Development
+\brief \htmlonly
+      <h3><span  style="text-decoration: underline;"><span  style="font-style: italic;"><span
+               style="color: rgb(102, 102, 102);">Skin Add-On Development</span></span></span></h3>
+       \endhtmlonly
+
+Kodi includes a GUI library that allows you to skin/change everything you see in
```
Comment (crash123) [inline xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/skin.dox]: Just a suggestion to break up your long sentences: Kodi includes a GUI library that allows you to skin/change everything you see. The images, sizes and positions of controls, colours, fonts, text, through to altering navigation and even adding new functionality.

File: xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/skin.dox [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+Kodi, from the images, the sizes and positions of all controls, colours, fonts,
+and text, through to altering navigation and even adding new functionality.
+
+The skin system is quite complex, and this portion of the manual is dedicated to
+providing in depth information on how it all works, along with tips to make the
+experience a little more pleasant.
+
+If you are just getting started with skinning Kodi, then it is suggested that the
+best way to learn is by modifying one of the many existing skins that are
+available. The default skin, Estuary, includes almost all the various tricks
```
Comment (crash123) [inline xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/skin.dox]: Superfluous comma: The default skin Estuary, includes...

Comment (garbear) [inline xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/skin.dox]: nope, comma is correct

File: xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/skin.dox (hunk 5 above)
Comment (AlwinEsch) [inline xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/skin.dox]: Thanks, this and the other is changed.

**Comment by NedScott** [comment]
Outside of skin, python add-ons, and some JSON-RPC related topics, our wiki doesn't really have a lot of developer documentation. It's mostly about user documentation. If it makes it easier for devs to use only doxygen then I'm all for it.

That being said, the wiki can easily hook into doxygen documentation. As in, wiki pages can be automatically updated when github is updated. I looked into it once, years ago, but didn't bother because we barely had any documentation in the code at that time ;)

The wiki can even transclude specific lines of code directly form github (we don't currently have this ability as we're waiting for the server/mediawiki to be updated).

So in that sense, the code/github can generate the core-documentation, then the wiki could be used for annotations, examples, and other detailed information. If that sounds like it might be useful then let me know and I will look into it.

**Comment by AlwinEsch** [comment]
Hi @NedScott you can read my mind's :D

My only new idea for this to improve it, is to change the places of the files like e.g. `xbmc/addons/kodi-addon-dev-kit/doxygen/Skin/controls/Button_control.txt` to `xbmc/guilib/GUIButtonControl.dox`. Then the text is more near on related code files. To include it direct there is it for me to big.

Are this inside the request OK for you?

**Comment by AlwinEsch** [comment]
Have changed the places to related source files, OK to merge?

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by NedScott** [comment]
Sounds good to me. I'll make a forum post about ideas for doxygen <-> wiki stuff.

**Comment by MartijnKaijser** [comment]
Doesn't touch core code

**Comment by AlwinEsch** [comment]
Thanks a lot!

**Comment by MartijnKaijser** [comment]
Small bug in this one https://codedocs.xyz/xbmc/xbmc/Radio_button_control.html
The "A" points to a class while it shouldn't

**Pull request merged**